src/main/java/org/opensearch/flowframework/workflow/WorkflowProcessSorter.java

```diff
@@ -244,11 +254,17 @@ public void validateGraph(List<ProcessNode> processNodes, WorkflowValidator vali
                 );
             }
         }
+    }
 
+    private WorkflowValidator readWorkflowValidator() throws IOException {
+        return WorkflowValidator.parse("mappings/workflow-steps.json");
     }
 
-    private TimeValue parseTimeout(WorkflowNode node) {
-        String timeoutValue = (String) node.userInputs().getOrDefault(NODE_TIMEOUT_FIELD, NODE_TIMEOUT_DEFAULT_VALUE);
+    private TimeValue parseTimeout(WorkflowNode node) throws IOException {
+        WorkflowValidator validator = readWorkflowValidator();
+        String nodeTimeoutValue = Optional.ofNullable(validator.getWorkflowStepValidators().get(node.type()).getTimeout())
+            .orElse(NODE_TIMEOUT_DEFAULT_VALUE);
+        String timeoutValue = (String) node.userInputs().getOrDefault(NODE_TIMEOUT_FIELD, nodeTimeoutValue);
```

Comment:
Part of me thinks we should avoid calculating the default if we don't need it.

The other part of me thinks that mot of the time the user won't provide an override and we will use the default, and working around it is premature optimization.  So I guess this is all fine as is.